pkg/execution/pauses/block.go

```diff
@@ -234,6 +272,17 @@ func (b blockstore) flushIndexBlock(ctx context.Context, index Index) error {
 		// from iterators often;  this only happens in a race where the iterator
 		// has pauses in-memory which are then deleted while iterating, leading to
 		// a race condition.
+
+		// Another case where this could happen is during rollout when we are not
+		// deleting pauses from the buffer after flushing to a block which will
+		// make it that we trigger flush jobs because the buffer is always at the
+		// threshold but we can't find enough pauses since the last block.
```

Comment:
would be nice to have some metrics on each type of scenario to perhaps help with identifying further optimizations later.